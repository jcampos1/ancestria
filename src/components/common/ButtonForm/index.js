import React from 'react';
import PropTypes from 'prop-types';

class ButtonForm extends React.PureComponent {
	render() {
		const {isLoading, name, nameLoading} = this.props;

		return (
			<React.Fragment>
			{
				isLoading ? ( 
					<button className="form-control btn btn-primary" disabled={true}>{nameLoading}...</button>
				) : (
					<button className="form-control btn btn-primary" type="submit">{name}</button>
				)
			}
			</React.Fragment>
		)
	}
}

ButtonForm.propTypes = {
	isLoading: PropTypes.bool.isRequired,
	name: PropTypes.string.isRequired,
	nameLoading: PropTypes.string.isRequired,
}

ButtonForm.defaultProps = {
	isLoading: false
}

export default ButtonForm;