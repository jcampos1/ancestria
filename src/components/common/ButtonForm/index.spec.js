import React from 'react';
import {shallow} from 'enzyme';
import ButtonForm from 'components/common/ButtonForm/index';

describe('ButtonForm component', () => {
	it('Should render ButtonForm component when is not loading', () => {
		const name = "Log in";
		const nameLoading = "Logging in";

		const wrapper = shallow(<ButtonForm name={name} nameLoading={nameLoading}/>);
		expect(wrapper).toMatchSnapshot();

		expect(wrapper.find("Fragment button[type='submit']").contains(name)).toEqual(true);
	});

	it('Should render ButtonForm component when is loading', () => {
		const isLoading = true;
		const name = "Log in";
		const nameLoading = "Logging in";

		const wrapper = shallow(<ButtonForm isLoading={isLoading} name={name} nameLoading={nameLoading}/>);
		expect(wrapper).toMatchSnapshot();

		expect(wrapper.find("Fragment button[disabled=true]").contains(nameLoading)).toEqual(true);
	});
});