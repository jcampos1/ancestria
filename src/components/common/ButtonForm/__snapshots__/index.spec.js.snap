// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonForm component Should render ButtonForm component when is loading 1`] = `
<Fragment>
  <button
    className="form-control btn btn-primary"
    disabled={true}
  >
    Logging in
    ...
  </button>
</Fragment>
`;

exports[`ButtonForm component Should render ButtonForm component when is not loading 1`] = `
<Fragment>
  <button
    className="form-control btn btn-primary"
    type="submit"
  >
    Log in
  </button>
</Fragment>
`;
